six/modules/c++/cphd/include/cphd/SupportBlock.h

```diff
@@ -0,0 +1,169 @@
+/* =========================================================================
+ * This file is part of cphd-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2019, MDA Information Systems LLC
+ *
+ * cphd-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef __CPHD_SUPPORT_BLOCK_H__
+#define __CPHD_SUPPORT_BLOCK_H__
+
+#include <iostream>
+#include <string>
+#include <complex>
+#include <unordered_map>
+
+#include <sys/Conf.h>
+#include <io/SeekableStreams.h>
+#include <types/RowCol.h>
+
+#include <mem/ScopedArray.h>
+#include <mem/SharedPtr.h>
+#include <mem/BufferView.h>
+
+#include <cphd/Data.h>
+#include <cphd/Utilities.h>
+
+namespace cphd
+{
+
+/*
+ *  \struct SupportBlock
+ *
+ *  \brief This class contains information about the SupportBlock CPHD data.
+ */
+//  Due to the large size of CPHD SupportBlock, this object does not contain
+//  any actual SupportBlock data
+class SupportBlock
+{
+public:
+    /*
+     *  \func SupportBlock
+     *
+     *  \brief Constructor initializes book keeping information
+     *
+     *  \param pathname Input CPHD pathname to initialize a file input stream
+     *  \param data Data section from CPHD
+     *  \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
+     *  \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
+     */
+    SupportBlock(const std::string& pathname,
+                 const cphd::Data& data,
+                 sys::Off_T startSupport,
+                 sys::Off_T sizeSupport);
+
+    /*
+     *  \func SupportBlock
+     *
+     *  \brief Constructor initializes book keeping information
+     *
+     *  \param inStream Input stream to an already opened CPHD file
+     *  \param data Data section from CPHD
+     *  \param startSupport CPHD header keyword "SUPPORT_BLOCK_BYTE_OFFSET"
+     *  \param sizeSupport CPHD header keyword "SUPPORT_BLOCK_SIZE"
+     */
+    SupportBlock(mem::SharedPtr<io::SeekableInputStream> inStream,
```

Comment:
Can you use `std::shared_ptr`? (If you have C++11 this does some fancy delegation/subclass work to get the same thing for backwards-compatability reasons, but since we require C++11 now, we should just be able to use the real thing.)

Reply:
Sure. I'll make this change to CPHD1.0 in general.